commands/snapshotautodeletecommand.php

```diff
@@ -0,0 +1,136 @@
+            if (!isset($this->config['api']['user']) || !isset($this->config['api']['password'])) throw new Exception("No user or no password configured to connect ProfitBricks!");
+        }
+        else throw new Exception("Error during reading config.ini!");
+        $this->profitBricksApi->setUserName($this->config["api"]["user"]);
+        $this->profitBricksApi->setPassword($this->config["api"]["password"]);
+
+        $snapshots = $this->profitBricksApi->snapshots();
+        if ($snapshots != false && count($snapshots)>0)
+        {
+            $io =  new SymfonyStyle($input, $output);
+            $io->title("Snapshots automatic deletion  ".date("d.m.Y H:i:s"));
+            $tableRows = array ();
+            $header = array ("ID", "Name", "Description", "Date       Time", "Size", "State");
+            $now = new DateTime();
+            $sumSize = 0;
+            $sumCount = 0;
+            $this->readVirtualMachinesOnce();
+            foreach ($snapshots as $snapshot)
+            {
+                if ($virtualMachine = $this->getVirtualMachineFor($snapshot)) // don't use ==, because it is an valid assignment
+                {
+                    if (array_key_exists($virtualMachine->name, $this->config)) // check if config exists
+                    {
+                        $deletionDate = clone $snapshot->createdDate;
+                        $deletionDate->add(new \DateInterval("P".$this->config[$virtualMachine->name]["deleteSnapshotsOlderThan"]."D"));
+                        if ($snapshot->autoScriptCreated && $now>=$deletionDate)
+                        {
+                            if ($this->profitBricksApi->deleteSnapshot($snapshot->id))
+                            {
+                                $tableRows[] = array ($snapshot->id, $snapshot->name, $snapshot->description, $snapshot->createdDate->format("d.m.Y H:i"), $snapshot->size." GB", "deleted!");
+                                $sumSize += (int)$snapshot->size;
+                                $sumCount += 1;
+                            }
+                            else $tableRows[] = array ($snapshot->id, $snapshot->name, $snapshot->description, $snapshot->createdDate->format("d.m.Y H:i"), $snapshot->size." GB", "deletion failed!");
+                        }
+                        else $tableRows[] = array ($snapshot->id, $snapshot->name, $snapshot->description, $snapshot->createdDate->format("d.m.Y H:i"), $snapshot->size." GB", "deletion time: ".$deletionDate->format("d.m.Y H:i"));
+                    }
+                    else  $tableRows[] = array ($snapshot->id, $snapshot->name, $snapshot->description, $snapshot->createdDate->format("d.m.Y H:i"), $snapshot->size." GB", "no configuration for server ".$virtualMachine->name."!");
+                }
+                else  $tableRows[] = array ($snapshot->id, $snapshot->name, $snapshot->description, $snapshot->createdDate->format("d.m.Y H:i"), $snapshot->size." GB", "no valid server name!");
```

Comment:
This bunch of else-cases also share a lot of redundant parts. A possibility of reducing redundancy here could be to only set the changing part (last array index) in the else-cases and finish the operation by setting:
`$tableRows[] = array ($snapshot->id, $snapshot->name, $snapshot->description, $snapshot->createdDate->format("d.m.Y H:i"), $snapshot->size." GB", $SET_BY_ELSE_CASE);`